internal/types/subtypes/attribute_transform.go

```diff
@@ -0,0 +1,186 @@
+package subtypes
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func convertAttributesToSubtype(msg proto.Message, st Subtype) error {
+	r := msg.ProtoReflect()
+	d := r.Descriptor()
+
+	defaultAttrField, err := attributeField(d, defaultSubtype)
+	if err != nil {
+		// no need to convert
```

Comment:
This comment is a little confusing, why, when there is an error, is there no need to convert?

Reply:
I can expand the comment to clarify. But `attribueField` will return an error if the given message is not registered, which will be the case for a message like [this test case](https://github.com/hashicorp/boundary/blob/fc2447052a6ddc85da8be422437f13675f5878aa/internal/types/subtypes/interceptor_test.go#L247-L271). So if we are not able to get a `defaultAttrField` there should be no need to convert.

Reply:
Yeah I think it would be valuable to expand the comment 👍🏻.

Reply:
See 52cf23ce7